src/pymmcore_widgets/_mda/_positions_table_widget.py

```diff
@@ -538,35 +538,35 @@ def set_state(
             By default True. If True, the current positions list is cleared before the
             specified one is added.
         """
-        if clear:
-            self.clear()
-
         if not isinstance(positions, Sequence):
             raise TypeError("The 'positions' arguments has to be a 'Sequence'.")
 
         if not self._mmc.getXYStageDevice() and not self._mmc.getFocusDevice():
             raise ValueError("No XY and Z Stage devices loaded.")
 
-        for position in positions:
-            if isinstance(position, Position):
-                position = cast("PositionDict", position.dict())
-
-            if not isinstance(position, dict):
-                continue
+        with signals_blocked(self):
```

Comment:
delaying the valueChanged signal till the very end is effectively the maximum speed up possible here.